### PR TITLE
Update convnet.py

### PR DESCRIPTION
**Pull request by AdityaSoni19031997**
@zishansami102 
added lecun_normal which uses gaussian distribution to initialise weights

### CODE REVIEW

**Comment by AdityaSoni19031997** [comment]
capture

File: MNIST/convnet.py [hunk 1]
```diff
@@ -143,6 +143,25 @@ def initialize_param(f, l):
...
+    if distribution not in {'normal'}:
+        raise ValueError('Invalid `distribution` argument: '
+                             'expected one of {"normal", "uniform"} '
+                             'but got', distribution)
+
+    scale = scale
+    distribution = distribution
+    fan_in = FILTER_SIZE*FILTER_SIZE*IMG_DEPTH
+    scale /= max(1., fan_in)
+    stddev = scale * np.sqrt(1./fan_in)
```
Comment (zishansami102) [inline MNIST/convnet.py]: As far as i have read about lecun_normal, It should have stddev of about np.sqrt(1./fan_in) & not (1./fan_in) * np.sqrt(1./fan_in) [given  by line no. 161 --  scale * np.sqrt(1./fan_in) where scale=(1./fan_in)]  which is in this case.

**Comment by zishansami102** [comment]
@AdityaSoni19031997 
As you can see that both of your PR has bugs. I cannot merge them.
Just because training has started does not guarantee that the code is bug free. Take your time until it starts to converge. Around 20% of the 1st epoch, it should have a decent accuracy.
& Be patient while writing code. ;)

**Comment by AdityaSoni19031997** [comment]
https://github.com/chainer/chainer/blob/v3.1.0/chainer/initializers/normal.py#L39

**Comment by AdityaSoni19031997** [comment]
i had multiple commits to fix the NUM_Filter one

Scale can be anything , if someone is happy with scale as `1` it can be modified

Done the modification

**Comment by zishansami102** [comment]
Did you make it converge?
Run the training, and wait to atleast 20-25% of the 1st epoch and paste the screenshot here. It should converge.

**Comment by zishansami102** [comment]
[For stddev. confusion](https://github.com/fchollet/keras/blob/master/keras/initializers.py#L196)
I don't know. But what i have read & what the code in the above link is showing is the same.
Maybe that's a little twisted model[your reference], i don't have any idea about that.

**Comment by AdityaSoni19031997** [comment]
capture
The accuracy is increasing and loss is going down so that's a sufficient indication of your code running like a charm...

**Comment by AdityaSoni19031997** [comment]
@zishansami102

Because i had seen that Keras does it that way..

**Comment by zishansami102** [comment]
Keras is different than your earlier code.
In Keras : stddev = np.sqrt(scale) where scale /= max(1. , fan_in)

**Comment by AdityaSoni19031997** [comment]
We should change the theta also?

**Pull request merged**